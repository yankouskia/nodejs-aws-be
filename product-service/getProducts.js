import { loadProducts } from './db';

export const getProducts = async event => {
  try {
    const products = await loadProducts();

    return {
      statusCode: 200,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
      },
      body: JSON.stringify({ products }),
    };
  } catch (e) {
    console.log('Database Error', e);

    return {
      statusCode: 500,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
      },
      body: JSON.stringify({ message: 'Error while reading data' }),
    };
  }
};
