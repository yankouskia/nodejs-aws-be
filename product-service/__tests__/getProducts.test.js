import { getProducts } from '../getProducts';

jest.mock('../db.js', () => ({
  loadProducts: () => ([{
    id: 'product',
  }])
}))

test('should return product', async () => {
  const product = await getProducts();
  expect(product.statusCode).toEqual(200);
  expect(product.body).toMatchSnapshot();
});
