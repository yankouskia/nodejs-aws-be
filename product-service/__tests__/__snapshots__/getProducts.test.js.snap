// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return product 1`] = `"{\\"products\\":[{\\"id\\":\\"product\\"}]}"`;
